aptos-move/aptos-vm/src/block_executor/mod.rs

```diff
@@ -159,6 +161,22 @@ impl BlockAptosVM {
             });
         drop(signature_verification_timer);
 
+        // The signatures are already verified, so we just have to filter out duplicated
+        // (sender, sequence_number).
+        let duplicates_timer = BLOCK_EXECUTOR_DUPLICATES_SECONDS.start_timer();
+        let mut seen = HashSet::new();
+        let mut num_duplicates: usize = 0;
+        for txn in &mut signature_verified_block {
+            if let PreprocessedTransaction::UserTransaction(inner) = txn {
+                if !seen.insert((inner.sender(), inner.sequence_number())) {
```

Comment:
Can we use the tx hash or signature instead? Both should be unique per distinct tx. Using account and nonce is fine for now but won’t work on nonce-less or account-less txs in the future

Reply:
With that said, we certainly need to consider the perf implication of doing so as both tx hash and signature would be longer => more bytes to check

Reply:
Another side note is that account + nonce do not guarantee dup txs as the contents can be different. This could happen from a multi-flient setup that uses the same account and send different txs with stale seq number

Reply:
@igor-aptos why does it break backwards compatibility? The txn should be discarded regardless. Is the discarded reason saved on chain?

Reply:
Ah yes, actually now I remember I had considered this originally. The previous approach of using the hash doesn't have this issue. 

For example if the first txn in the block is discarded or failed for a different reason after the signature check, then the second txn should still execute. But with this implementation that doesn't happen. It's actually the main reason behind the unit test failures (with some nuance).

Reply:
Yeah this is why I would suggest using tx hash or signature. It's technically not correct to only look at account + nonce to dedup txs

Reply:
signature caching might be an interesting idea to save on hashing, but also it was always interesting to me that we didn't seem to cache the hash of the txn, that's for pre-computation required.

De-dup can be bucketed based on sender or sender / seq num though, since we are concerned about false positives here for dedup. That makes the actual de-dup process quite easy to parallelize. Otherwise, we had tried some direct approaches for deduplicating before execution with Shir and @zjma and it wasn't straightforward, but even that is surely doable, just maybe more involved than using some crate directly.

I also didn't find the original approach too complicated and happy to approve if we go back to it and iterate gradually.